base_layer/core/src/base_node/states/block_sync.rs

```diff
@@ -20,46 +20,108 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-use crate::base_node::states::{
-    fetching_horizon_state::HorizonInfo,
-    InitialSync,
-    ListeningInfo,
-    StateEvent,
-    StateEvent::FatalError,
+use crate::{
+    base_node::{
+        base_node::BaseNodeStateMachine,
+        states::{fetching_horizon_state::HorizonInfo, InitialSync, ListeningInfo, StateEvent},
+    },
+    chain_storage::{BlockchainBackend, ChainMetadata},
 };
 use log::*;
 
 const LOG_TARGET: &str = "base_node::block_sync";
 
+// The number of Blocks that can be requested in a single query from remote nodes.
+const BLOCK_SYNC_CHUNK_SIZE: usize = 2;
```

Comment:
I think this should rather be what a node is prepared to _send_ in a single chunk. Otherwise it feels like a fairly trivial DOS attack to set this very high (say 1,000) and make simultaneous requests to a victim node from several attacker nodes

Reply:
I agree, this is an issue. It seems as if the problem is with the Base Node Service fetch block request and response that is providing as many blocks as were requested. I think it should be changed so that a node can request as many as it wants, but the responding base node will only provided a minimum of one block up to the limit that was internally configured. The requested blocks over the limit will then be ignored and must be requested again. This block sync code will then have to changed slightly as it wont necessary receive all the blocks it requested.

Reply:
Sounds good